src/here-xyz.ts

```diff
@@ -1490,222 +1311,7 @@ function taskQueue(size: number = 8, totalTaskSize: number) {
-                options,
-                tags,
-                object,
-                true,
-                options.ptag,
-                options.file,
-                options.id
-            );
-
-        } else {
-            if (!options.stream) {
-                let result = await transform.read(
-                    options.file,
-                    false
-                );
-                let object = JSON.parse(result);
-                if(!(object.features && object.features.length > 0 && object.features[0].type == 'Feature')){
-                    object = {
-                        features: await transform.transform(
-                            object,
-                            options
-                        ),
-                        type: "FeatureCollection"
-                    };
-                }
-                await uploadData(
-                    id,
-                    options,
-                    tags,
-                    object,
-                    true,
-                    options.ptag,
-                    options.file,
-                    options.id
-                );
-            } else {
-                let queue = streamingQueue();
-                let c = 0;
-                await transform.readGeoJsonAsChunks(options.file, options.chunk ? options.chunk : 1000, options, async function (result: any) {
-                    if (result.length > 0) {
-                        const fc = {
-                            features: result,
-                            type: "FeatureCollection"
-                        };
-                        await queue.send({ id: id, options: options, tags: tags, fc: fc, retryCount: 3 });
-                    }
-                    return queue;
-                });
-                while (queue.chunksize != 0) {
-                    await new Promise(done => setTimeout(done, 1000));
-                }
-            }
-        }
-    } else {
-        const getStdin = require("get-stdin");
-        await getStdin().then((str: string) => {
-            try {
-                const obj = JSON.parse(str);
-                uploadData(
-                    id,
-                    options,
-                    tags,
-                    obj,
-                    false,
-                    options.ptag,
-                    null,
-                    options.id
-                );
-            } catch (e) {
-                console.log(
-                    "Empty or invalid input to upload. Refer to 'here xyz upload -h' for help"
-                );
-                process.exit(1);
-            }
-        });
-    }
-
-    let totalTime = ((new Date().getTime() - startTime.getTime()) / 1000);
-    console.log(options.totalCount + " features uploaded to XYZ space '" + id + "' in " + totalTime + " seconds, at the rate of " + Math.round(options.totalCount / totalTime) + " features per second");
-    //})();
+    return await xyzComm.uploadToXyzSpace(id, options);
```

Comment:
same as before comment, please dont create method that just redirects the work to other method

Reply:
Update - I have refactored most of the functions which were a redirect to another method, However some functions including uploadToXyzSpace are linked in tests directory. Moving these methods failed those tests hence had to wrap these functions around it.